### DISCUSSION TITLE
Gateway pods keeps restarting after upgrade

### DISCUSSION THREAD

**Discussion by edwardzjl** [istio/istio#54083] [Q&A]
I recently upgraded my istio to 1.24.1 using helm (revisioned). After the upgrade, all the gateway (`gateways.gateway.networking.k8s.io`) pods (including gateway and waypoint) keeps restarting. Here's partial log of istiod:

```console
...
2024-11-27T09:25:44.241021Z	info	gateway	reconciling	gateway=jlzhou-chatbot/waypoint
...
2024-11-27T09:25:44.428932Z	info	gateway	gateway updated	gateway=jlzhou-chatbot/waypoint
```

I have no idea why they are reconciled.

**Comment by howardjohn** [comment]
Maybe you can do `kubectl get deployment -ojson -w`  (and also on `service` and `gtw` maybe) to see the changes?

https://github.com/howardjohn/kubectl-grep?tab=readme-ov-file#diffing can also help diff it for you

**Comment by edwardzjl** [comment]
After I cleaned up the old version (1.24.0) the restarting stopped.
